### PR TITLE
DOC: update development documentation for link to built docs #37248

### LINKED ISSUE #37248: DOC: update development documentation for link to built docs
this has the built docs: https://pandas.pydata.org/pandas-docs/dev/ rather than https://dev.pandas.io/, so need to update the contributing docs.

### PR DESCRIPTION
**Pull request by aniaan**
- [x] closes #37248

cc: @jreback

### CODE REVIEW

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -598,7 +598,7 @@ Building master branch documentation
 
 When pull requests are merged into the pandas ``master`` branch, the main parts of
 the documentation are also built by Travis-CI. These docs are then hosted `here
-<https://dev.pandas.io>`__, see also
+<https://pandas.pydata.org/pandas-docs/dev/>`__, see also
```
Comment (jorisvandenbossche) [inline doc/source/development/contributing.rst:601]:
```suggestion
<https://pandas.pydata.org/docs/dev/>`__, see also
```

(this shorter version also works)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR!

File: doc/source/development/contributing.rst (hunk 1 above)
Comment (jreback) [reply doc/source/development/contributing.rst:601]: the shorter version does *not* seem to work (longer does)

**Comment by BeanNan** [comment]
I see that both versions can work normally, so I chose the shorter version

File: doc/source/development/contributing.rst (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/development/contributing.rst]: We use the shorter version in several places (eg the README, the issue template, etc). Do you have a link that doesn't work?

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @BeanNan !

**Pull request merged**